server/internal/saves/index.ts

```diff
@@ -62,7 +62,7 @@ class SaveManager {
     await Promise.all([hashPromise, uploadStream]);
 
     if (!hash) {
-      await objectHandler.delete(newSaveObjectId);
+      await objectHandler.deleteWithPermission(newSaveObjectId, userId);
```

Comment:
This need to bypass permissions, the user doesn't have access to directly delete this object (which is what this checks)